specs/phase0/p2p-interface.md

```diff
@@ -391,11 +392,11 @@ The `ErrorMessage` schema is:
 
 ```
 (
-  error_message: String
+  error_message: List[byte, 256]
 )
 ```
 
-*Note*: The String type is encoded as UTF-8 bytes without NULL terminator when SSZ-encoded. As the `ErrorMessage` is not an SSZ-container, only the UTF-8 bytes will be sent when SSZ-encoded.
```

Comment:
How were people encoding this `String` before? What limit are clients using?

Just trying to understand how we're dealing with this today to better understand how to treat this change in a release (e.g. how breaking is it)

Reply:
One could say that the `List` type is an attempt to preserve some semblance of backwards compatibility over `Bytes32` for example which would have a different encoding on the wire. I'm not sure what other clients used, we simply allowed any length.

The changes on BlocksByXxx are less of an issue since clients probably support lists longer than whatever requests they make - the only trouble would come from someone sending a `BlocksByRoot` request with more than 1024 hashes in it.

Reply:
yeah cool. 

I suppose the only issue with the `String` is if some clients were picking a max at random.

should be good to go. thanks!

Reply:
There's another minor incompatibility btw: the previous spec hints at utf-8 being required - this could also be considered a breaking change.